src/Operations/SHA3.py

```diff
@@ -0,0 +1,43 @@
+from Operation import Operation
+
+
+class SHA3(Operation):
+    def proceed(self, state):
+
+        offset = state.stack.pop().get_low_bits()
+        length = state.stack.pop().get_low_bits()
+
+        output = state.request_fresh_name()
+        state.stack.push_ref(output)
+
+        return [
+            f"""
+                let (local msize) = update_msize(msize, {offset}, {length})
+                let (local memval) = create_from_memory({offset}, {length})
+                local high
+                local low
+                %{{
+                    from Crypto.Hash import keccak
+
+                    keccak_hash = keccak.new(digest_bits=256)
+                    arr = []
+                    arr_length = {length}//16 if {length} % 16 == 0 else {length}//16+1
```

Comment:
Just a note: I don't know if you consider it more readable, but ceil division can be expressed as `({length} - 1) // 16 + 1`